api/address.proto

```diff
@@ -8,14 +9,23 @@ message Address {
     // For listeners, bind to 0.0.0.0 or :: to support the use of prefix/suffix
     // matching in FilterChainMatch. For clusters, an address may be either an
     // IP or hostname to be resolved via DNS.
-    google.protobuf.StringValue address = 1;
+    string address = 1;
     google.protobuf.UInt32Value port = 2;
   }
   message Pipe {
-    google.protobuf.StringValue path = 1;
+    string path = 1;
+  }
+  // Support pluggable resolvers.
+  message ResolverAddress {
+    // Name of the resolver. This must have been registered with Envoy.
```

Comment:
@PiotrSikora The point of this PR is to start a discussion about this API. We will need it at Google, we can discuss offline if you would like.